packages/interactivity-router/src/index.ts

```diff
@@ -108,20 +129,51 @@ const regionsToVdom: RegionsToVdom = async ( dom, { vdom, url } = {} ) => {
 		Promise.all( preloadScriptModules( dom ) ),
 	] );
 
-	return { regions, styles, scriptModules, title, initialData, url };
+	return {
+		regions,
+		regionsToAttach,
+		styles,
+		scriptModules,
+		title,
+		initialData,
+		url,
+	};
 };
 
 // Render all interactive regions contained in the given page.
 const renderRegions = ( page: Page ) => {
 	applyStyles( page.styles );
 
+	// Clone regionsToAttach.
+	const regionsToAttach = { ...page.regionsToAttach };
```

Comment:
This object is cloned because it's modified later on, but we don't want to alter the original object.